src/handler.js

```diff
@@ -39,18 +39,42 @@ const getReplyToFromRequestHeaders = (request) => {
 }
 
 const createFSPIOPErrorFromErrorResponse = (request, response) => {
-  const fspiopError = ((httpStatusCode) => {
-    switch (httpStatusCode) {
-      case 400:
-        return Errors.CLIENT_ERROR
-
-      case 404:
-        return Errors.UNKNOWN_URI
+  // The content of this function is a naive attempt at encoding a complex set of conditions
+  // declaratively, without good knowledge of future requirements. Put another way: if this becomes
+  // unwieldly, there's no reason it shouldn't be changed.
+  const getBadRequestMessage = inputMessage => {
+    const badRequestMessages = [
+      {
+        // Look for invalid headers and return a MALFORMED_SYNTAX error
+        regex: /^Invalid.*header$/,
```

Comment:
Should we not ignore case here?

Reply:
Note that this is not just for headers. This also is will occur on validations for the payload, params etc.

Reply:
> Note that this is not just for headers. This also is will occur on validations for the payload, params etc.

Actually, this should be handled by JOI. Ignore this comment.